backend/app/services/qualtrics_launcher.rb

```diff
@@ -3,15 +3,17 @@
 require 'openssl'
 
 class QualtricsLauncher
+  include Rails.application.routes.url_helpers
 
-  def initialize(config:, user_id:, study_id:)
-    @config = config
+  def initialize(secret_key:, survey_id:, user_id:, study_id:)
+    @secret_key = secret_key
+    @survey_id = survey_id
     @user_id = user_id
     @study_id = study_id
   end
 
   def url
-    uri = URI(config[:url])
+    uri = URI("#{Rails.application.secrets.qualtrics_launch_url}/#{survey_id}")
```

Comment:
This does lock us in on using our qualtrics (or at least, only 1 qualtrics account) for the entire app. The previous "give us the whole url" did allow for using arbitrary qualtrics accounts. Fine for now, but perhaps we should use the launch url from secrets/config as the default for an optional initialization param, so some surveys can have different accounts in the future.

Reply:
good catch 😁 

Yes, I thought of that, but decided against it for right now.  in the future when we use more of the qualtrics api we'll probably be using oauth with it, because that will allow a teacher to grant us access to their account by redirecting to qualtrics and granting us access.

When that happens we'll need a "qualtrics" profile associated with researchers and will no longer use the account specific urls at all.

Reply:
👍🏻